### PR TITLE
Initial documentation to Java

### PR DESCRIPTION
**Pull request by otaviojava**
:fire: https://github.com/platformsh/language-examples/pull/23

### CODE REVIEW

File: src/languages/java.md [hunk 1]
```diff
@@ -1,3 +1,101 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is the most popular language in the globe, and it is in several fields:
+
+* Mobile applications
```
Comment (chadwcarlson) [inline src/languages/java.md]: Is this list necessary here? We don't have something similar for the other languages, just a one or two sentence description.

File: src/languages/java.md [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+To select a Java version, specify a `type` such as `java:11`:
+
+```yaml
+# .platform.app.yaml
+type: "java:11"
+```
+
+## Support libraries
+
+While it is possible to read the environment directly from your application, it is generally easier and more robust to use the [`platformsh/config-reader`](https://github.com/platformsh/config-reader-java) Composer library which handles decoding of service credential information for you.
```
Comment (chadwcarlson) [inline src/languages/java.md]: Composer?

File: src/languages/java.md [hunk 3]
```diff
@@ -1,3 +1,101 @@
...
+
+## Support build automation
+
+Platform.SH has support for software project management and comprehension tool in the Java World:
```
Comment (chadwcarlson) [inline src/languages/java.md]: Platform.sh

File: src/languages/java.md [hunk 4]
```diff
@@ -1,3 +1,101 @@
...
+## Support build automation
+
+Platform.SH has support for software project management and comprehension tool in the Java World: 
+
+* [Gradle](https://gradle.org/)
+* [Maven](https://maven.apache.org/)
+* [Ant](https://ant.apache.org/)
+
+
+## Accessing services
```
Comment (chadwcarlson) [inline src/languages/java.md]: Won't be able to do pull-ins until java is added to examples.docs.platform.sh, so we'll need service access examples here, obviously.

File: src/languages/java.md [hunk 5]
```diff
@@ -1,3 +1,101 @@
...
+
+To access various [services](/configuration/services.md) with PHP, see the following examples.  The individual service pages have more information on configuring each service.
```
Comment (chadwcarlson) [inline src/languages/java.md]: Also still says PHP not Java

File: src/languages/java.md (hunk 1 above)
Comment (otaviojava) [reply src/languages/java.md]: better?

File: src/languages/java.md (hunk 5 above)
Comment (otaviojava) [reply src/languages/java.md]: ops, thank you

File: src/languages/java.md (hunk 4 above)
Comment (otaviojava) [reply src/languages/java.md]: I'll comment that, but I believe that need to be there together with other languages.

File: src/languages/java.md [hunk 6]
```diff
@@ -1,3 +1,94 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is the most popular language in today’s world billion of devices running on a Java platform; Java touched every field of development of software, java’s features make it the most powerful language in the current market of software development. The distributed application also quickly developed with the help of java. In current time lots of java’s framework presents which makes it more potent like Spring, Jakarta EE, Hibernate, JSP, Servlet, Spring Boot. Enterprise application development always preferred with java language only because of security and scalability features of the Java. For sure, what makes Java the most powerful platform in the globe is not the technology, but the vibrant and helpful community.
```
Comment (chadwcarlson) [inline src/languages/java.md]: To me this still seems long for what it's trying to accomplish. Maybe something as short as:

Java is a general-purpose programming language, and one of the most popular in the world today. Platform.sh supports Java runtimes that can be used with build management tools such as Gradle, Maven, and Ant. 

I defer to @Crell.

Comment (otaviojava) [reply src/languages/java.md]: Thank you, I'll paste this.

File: src/languages/java.md [hunk 7]
```diff
@@ -1,3 +1,101 @@
...
+<!--
+
+{% language name="RabbitMQ", type="php", url="https://examples.docs.platform.sh/php/rabbitmq" -%}
+
+{% language name="Solr", type="php", url="https://examples.docs.platform.sh/php/solr" -%}
+
+{% codetabs name="Elasticsearch", type="php", url="https://examples.docs.platform.sh/php/elasticsearch" -%}
+
+{% language name="Memcached", type="php", url="https://examples.docs.platform.sh/php/memcached" -%}
+-->
```
Comment (Crell) [inline src/languages/java.md]: Commented block can be removed.

File: src/languages/java.md [hunk 8]
```diff
@@ -1,3 +1,101 @@
...
+
+
+
+{% language name="MongoDB", type="php", url="https://examples.docs.platform.sh/php/mongodb" -%}
+
+{% language name="MySQL", type="php", url="https://examples.docs.platform.sh/php/mysql" -%}
+
+{% language name="PostgreSQL", type="php", url="https://examples.docs.platform.sh/php/postgresql" -%}
+
+{% language name="Redis", type="php", url="https://examples.docs.platform.sh/php/redis" -%}
```
Comment (Crell) [inline src/languages/java.md]: Shouldn't these link to Java now? :smile:

File: src/languages/java.md [hunk 9]
```diff
@@ -1,3 +1,101 @@
...
+
+
+## Supported versions
+
+* 8
+* 11
+* 12
+
+
+To select a Java version, specify a `type` such as `java:11`:
```
Comment (Crell) [inline src/languages/java.md]: Somewhere in here we should specify that we run OpenJDK specifically.  I'm... not sure of the best way to phrase that.

File: src/languages/java.md [hunk 10]
```diff
@@ -1,3 +1,101 @@
...
+* Apache TomEE
+* Payara
+* Thorntail
+
+#### Visual sample
+Spring MVC
+Jakarta EE and MVC SPEC
+
+
+-->
```
Comment (Crell) [inline src/languages/java.md]: Why the big commented out block?  If those aren't ready just don't include them at all yet.

File: src/languages/java.md (hunk 7 above)
Comment (otaviojava) [reply src/languages/java.md]: I want to keep this on someplace to don't forget.
Do we have Jira? Can we use Github Project?

File: src/languages/java.md (hunk 8 above)
Comment (otaviojava) [reply src/languages/java.md]: ops :)

File: src/languages/java.md (hunk 10 above)
Comment (otaviojava) [reply src/languages/java.md]: Done

File: src/languages/java.md [hunk 11]
```diff
@@ -1,3 +1,62 @@
 # Java
 
-Java support is in Alpha and not generally available. Contact sales [https://platform.sh/contact](https://platform.sh/contact) if you are interested in running Java based applications on Platform.sh.
+Java is a general-purpose programming language, and one of the most popular in the world today. Platform.sh supports Java runtimes that can be used with build management tools such as Gradle, Maven, and Ant.
+
+
+## Supported versions
+
```
Comment (robertDouglass) [inline src/languages/java.md]: OpenJDK versions:

File: src/languages/java.md (hunk 9 above)
Comment (robertDouglass) [reply src/languages/java.md]: Just put it as a line above the version numbers.

Comment (otaviojava) [reply src/languages/java.md]: Done

File: src/languages/java.md (hunk 11 above)
Comment (otaviojava) [reply src/languages/java.md]: Done, what do you think?

File: src/languages/java.md [hunk 12]
```diff
@@ -1,3 +1,64 @@
...
+```
+
+
+## Support libraries
+
+While it is possible to read the environment directly from your application, it is generally easier and more robust to use the [`platformsh/config-reader`](https://github.com/platformsh/config-reader-java) which handles decoding of service credential information for you.
+
+## Support build automation
+
+Platform.sh has support for software project management and comprehension tool in the Java World:
```
Comment (Crell) [inline src/languages/java.md]: "Platform.sh supports the most common project management tools in the Java ecosystem, including:"

Comment (otaviojava) [reply src/languages/java.md]: Done

File: src/languages/java.md [hunk 13]
```diff
@@ -1,3 +1,64 @@
...
+
+
+
+## Accessing services
+
+To access various [services](/configuration/services.md) with Java, see the following examples.  The individual service pages have more information on configuring each service.
+
+
+
+{% language name="MongoDB", type="java", url="https://examples.docs.platform.sh/java/mongodb" -%}
```
Comment (Crell) [inline src/languages/java.md]: The first tag needs to be called "codetabs".  Only the subsequent ones are called "language".

Comment (otaviojava) [reply src/languages/java.md]: thank you

**Comment by Crell** [comment]
This can merge once https://github.com/platformsh/language-examples/pull/23 gets merged.  And we figure out why it's not building...

**Comment by otaviojava** [comment]
Moves the code to just one PR: 
https://github.com/platformsh/platformsh-docs/pull/1107/

**Pull request closed without merging**